### PR TITLE
Takeaway Weekend Challenge - Alina

### PR DESCRIPTION
**Pull request by AlinaGoaga**

### CODE REVIEW

File: lib/message.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  def send
+    message = @twilio_client.messages.create(
+      body: "Your order has been placed!
+        It will be with you at: #{Time.now + 3600}",
+      to: "+447427750432",
+      from: "+441543624591")
+    message.error_code.zero?
+  end
+
+end
```
Comment (Sindex42) [inline lib/message.rb]: Nice job getting the message to send! I used the dotenv gem demonstrated in [this tutorial](https://www.twilio.com/blog/2017/01/how-to-set-environment-variables.html)  to hide the sensitive user data

File: lib/takeaway.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  # to be continued: implement a method that manages all the below?
+  # def manage_order
+  #   order_from_menu
+  #   order.total_to_pay
+  #   send_message
+  # end
+
+  def order_from_menu
+    display_menu.each do |item, _|
```
Comment (Sindex42) [inline lib/takeaway.rb]: Nice use of a not equal statement below. I'm unsure what the underscore between the pipes does though?

File: spec/message_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  subject(:message) { described_class.new(twilio_client_double) }
+
+  it 'returns true when the message is sent' do
+    allow(twilio_client_double).to receive_message_chain(:messages, :create).and_return(twilio_response_double)
+    allow(twilio_response_double).to receive(:error_code).and_return(0)
+    expect(message.send).to eq true
+  end
+
+  it 'returns false when there is an error_code' do
+    allow(twilio_client_double).to receive_message_chain(:messages, :create).and_return(twilio_response_double)
```
Comment (Sindex42) [inline spec/message_spec.rb]: Nice mocking here! Would it have been more DRY to move this line to the top as it is repeated?

File: README.md [hunk 4]
```diff
@@ -1,79 +1,38 @@
...
+```
+- used rspec to write unit tests
+- classes being test in isolation with the help of doubles
+- to run tests simply run rspec in the main project folder 
 
-You can see your [test coverage](https://github.com/makersacademy/course/blob/master/pills/test_coverage.md) when you run your tests.
+```
+Application as of December 9th
+```
+![Image-of-application](https://github.com/AlinaGoaga/takeaway-challenge/blob/master/takeaway%20dec%209.jpeg)
```
Comment (stonefarmer9) [inline README.md]: Nice readme, maybe consider adding user stories ?

File: README.md [hunk 5]
```diff
@@ -1,79 +1,38 @@
...
-As a customer
-So that I can verify that my order is correct
-I would like to check that the total I have been given matches the sum of the various dishes in my order
-
-As a customer
-So that I am reassured that my order will be delivered on time
-I would like to receive a text such as "Thank you! Your order was placed and will be delivered before 18:52" after I have ordered
+First diagram
 ```
+![Image-of-application](https://github.com/AlinaGoaga/takeaway-challenge/blob/master/draft.jpg)
```
Comment (CelineKaslin) [inline README.md]: can't see the image of the diagram but it is a nice idea to incorporate them

File: README.md [hunk 6]
```diff
@@ -1,79 +1,38 @@
...
+```
+![Image-of-application](https://github.com/AlinaGoaga/takeaway-challenge/blob/master/final.jpg)
 
+```
+Code
+```
+The code is structured in 5 classes: Takeaway (main class), Order, Menu, User_input (for placing the order) and Message (for the notification the user gets once the order has been placed) 
 
-In code review we'll be hoping to see:
+The main class allows the user to see the menu, add items to the order (delegated to the UserInput and Order classes) and see the total to pay (again delegated to Order). It also delegates the sending of confirmation messages to the Messages class which performs the action via the Twilio API.
```
Comment (CelineKaslin) [inline README.md]: great to explain your approach :)

File: README.md [hunk 7]
```diff
@@ -1,79 +1,38 @@
...
-I would like to be able to select some number of several available dishes
-
-As a customer
-So that I can verify that my order is correct
-I would like to check that the total I have been given matches the sum of the various dishes in my order
-
-As a customer
-So that I am reassured that my order will be delivered on time
-I would like to receive a text such as "Thank you! Your order was placed and will be delivered before 18:52" after I have ordered
+First diagram
```
Comment (joaoag) [inline README.md]: Love having the hand drawn diagrams, they look great and help illustrate how everything fits together +1

File: lib/takeaway.rb (hunk 2 above)
Comment (stonefarmer9) [reply lib/takeaway.rb]: it indicates a non used parameter

File: README.md (hunk 6 above)
Comment (joaoag) [inline README.md]: Clear summary, great.

File: lib/menu.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+class Menu
+
+  def display
+    { "steak" => 10, "ravioli" => 20, "augbergine salad" => 5 }
```
Comment (joaoag) [inline lib/menu.rb]: This is a very specialised restaurant... 
: b

File: lib/takeaway.rb (hunk 2 above)
Comment (CelineKaslin) [inline lib/takeaway.rb]: I miss knowledge about the _ in the pipe, will read about what it is doing exactly

File: lib/takeaway.rb [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  def display_menu
+    menu.display
+  end
+
+  # to be continued: implement a method that manages all the below?
+  # def manage_order
+  #   order_from_menu
+  #   order.total_to_pay
+  #   send_message
+  # end
```
Comment (CelineKaslin) [inline lib/takeaway.rb]: nice futur encapsulation planned :)

File: README.md (hunk 6 above)
Comment (joaoag) [inline README.md]: Would you have considered having a display class for anything that's user facing? Or is that role fulfilled by restaurant class?

File: lib/user_input.rb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+class UserInput
+
```
Comment (CelineKaslin) [inline lib/user_input.rb]: great split of responsibility to have a separate class for the user input

File: lib/order.rb [hunk 11]
```diff
@@ -0,0 +1,24 @@
+require_relative 'menu'
```
Comment (joaoag) [inline lib/order.rb]: Nice short methods +1

File: lib/takeaway.rb [hunk 12]
```diff
@@ -0,0 +1,47 @@
+require_relative 'menu'
+require_relative 'user_input'
+require_relative 'order'
+require_relative 'message'
+
+class Takeaway
+
+  attr_reader :menu, :order, :user_input, :message
+
+  def initialize(menu = Menu.new, user_input = UserInput.new,
```
Comment (joaoag) [inline lib/takeaway.rb]: Great use of injection

**Pull request closed without merging**